### PR TITLE
feat(azure): New Azure checks related to CosmosDB

### PR DESCRIPTION
**Pull request by pedrooot**
### Context

New checks related to CosmosDB

### Description

Checks related to cosmosdb:

- [x] cosmosdb_account_firewall_use_selected_networks
- [x] cosmosdb_account_use_private_endpoints
- [x] cosmosdb_account_use_aad_and_rbac

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/azure/services/cosmosdb/cosmosdb_account_use_private_endpoints/cosmosdb_account_use_private_endpoints.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+            for account in accounts:
+                report = Check_Report_Azure(self.metadata())
+                report.subscription = subscription
+                report.resource_name = account.name
+                report.resource_id = account.id
+                report.status = "FAIL"
+                report.status_extended = f"CosmosDB account {account.name} from subscription {subscription} is not using private endpoints connections"
+                if account.private_endpoint_connections and any(
+                    account.private_endpoint_connections
+                ):
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/cosmosdb/cosmosdb_account_use_private_endpoints/cosmosdb_account_use_private_endpoints.py]:
```suggestion
                if account.private_endpoint_connections:
```

File: prowler/providers/azure/services/cosmosdb/cosmosdb_service.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+class Account:
+    id: str
+    name: str
+    kind: str
+    location: str
+    type: str
+    tags: dict
+    is_virtual_network_filter_enabled: bool
+    private_endpoint_connections: list[PrivateEndpointConnection] = None
+    disable_local_auth: bool = None
```
Comment (MrCloudSec) [inline prowler/providers/azure/services/cosmosdb/cosmosdb_service.py]:
```suggestion
    disable_local_auth: bool = False
```

File: pyproject.toml [hunk 3]
```diff
@@ -29,6 +29,7 @@ alive-progress = "3.1.5"
 awsipranges = "0.3.3"
 azure-identity = "1.15.0"
 azure-mgmt-authorization = "4.0.0"
+azure-mgmt-cosmosdb = "^9.4.0"
```
Comment (MrCloudSec) [inline pyproject.toml]:
```suggestion
azure-mgmt-cosmosdb = "9.4.0"
```

**Review by MrCloudSec — CHANGES_REQUESTED** [review]
Please, review my comments.

File: prowler/providers/azure/services/cosmosdb/cosmosdb_account_use_private_endpoints/cosmosdb_account_use_private_endpoints.py (hunk 1 above)
Comment (pedrooot) [reply prowler/providers/azure/services/cosmosdb/cosmosdb_account_use_private_endpoints/cosmosdb_account_use_private_endpoints.py]: Done!

File: prowler/providers/azure/services/cosmosdb/cosmosdb_service.py (hunk 2 above)
Comment (pedrooot) [reply prowler/providers/azure/services/cosmosdb/cosmosdb_service.py]: Done!

File: pyproject.toml (hunk 3 above)
Comment (pedrooot) [reply pyproject.toml]: Done!

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**